app.rb

```diff
@@ -0,0 +1,102 @@
+require_relative 'person'
+require_relative 'teacher'
+require_relative 'rental'
+require_relative 'book'
+require_relative 'students'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  # Method to list all books
+  def list_books
+    puts 'List of all books:'
+    @books.each do |_book|
+      puts "#{book.title} by {book.author}"
+    end
+  end
+
+  # Method to list all people
+  def list_people
+    puts 'List of all people:'
+    @people.each do |_person|
+      puts "#{person.name} (#{person.class})"
+    end
+  end
+
+  # Method to create a person
+  def create_person
+    puts 'Enter the name of the person:'
+    name = gets.chomp
+    puts 'Is this person a student or teacher? (s/t)'
+    type = gets.chomp.downcase
+
+    case type
+    when 's'
+      @people << Students.new(name)
+      puts 'Student {name} has been created.'
+    when 't'
+      @people << Teacher.new(name)
+      puts "Teacher #{name} has been created."
+    else
+      puts 'Invalid input. Please try again.'
+    end
+  end
+
+  # Method to create a book
+  def create_book
+    puts 'Enter the title of the book:'
+    @title = gets.chomp
+    puts 'Enter the author of the book:'
+    @author = gets.chomp
+    book << Book.new(title, author)
```

Comment:
- [ ] it seems that you have an typo here it should be `@books` not book I got the following error when I try to create a book so please make sure to fix that

Reply:
Hi @youmari 
Thank you for your detailed review.
You may wish to know that those are not just comments. I have disabled linter checks for line length since there's no any other way of checking for user input than to have all those lines of code.